### ISSUE TITLE
Problem colouring mesh by cells when n. of faces == n. of vertices 

### ISSUE THREAD

**Issue by kymok** [pyvista/pyvista#1442]
**Describe the bug, what's wrong, and what you expect:**

Some meshes such as a tetrahedron have same number of vertices and faces. Vertex shading is always preferred over cells in such a case but user should be able to choose which method to be preferred.

In the meantime I'm using some dirty workaround like adding a redundant vertex to vertices, but would be glad if this behaviour can be user configured.

Maybe related:  https://github.com/pyvista/pyvista/blob/9def273bf7df5f51796173148f8d729bed64409e/pyvista/plotting/plotting.py#L1636 

-----

**To Reproduce**

```py
# Construct some mesh
# 4 verts, 4 faces
vertices = np.array([[0, 0, 0], [1, 0, 0], [1, 1, 0], [0, 0, 1]])
faces = np.hstack([[3, 0, 1, 2], [3, 0, 3, 2], [3, 0, 1, 3], [3, 1, 2, 3]])
mesh = pyvista.PolyData(vertices, faces)
colors = [
    [255, 0, 0],
    [0, 255, 0],
    [0, 0, 255],
    [255,255,255]
]

mesh.cell_arrays['colors'] = colors 
mesh.plot(scalars='colors', show_edges=True, rgb=True)
```

**Screenshots**
Above: what we get, below: what I want to achieve.

<img width="521" alt="image" src="https://user-images.githubusercontent.com/28852446/123427417-8df5b900-d5ff-11eb-8884-910faf172925.png">

-----

**System Information:**

Running ubuntu 20.04 on Docker Desktop for Mac 3.3.3 (64133)

```txt
--------------------------------------------------------------------------------
  Date: Fri Jun 25 12:53:47 2021 UTC

                OS : Linux
            CPU(s) : 4
           Machine : x86_64
      Architecture : 64bit
       Environment : Jupyter
        GPU Vendor : Mesa/X.org
      GPU Renderer : llvmpipe (LLVM 11.0.0, 256 bits)
       GPU Version : 4.5 (Core Profile) Mesa 20.2.6

  Python 3.8.5 (default, May 27 2021, 13:30:53)  [GCC 9.3.0]

           pyvista : 0.31.3
               vtk : 9.0.1
             numpy : 1.21.0
           imageio : 2.9.0
           appdirs : 1.4.4
            scooby : 0.5.7
            meshio : 4.4.6
        matplotlib : 3.4.2
           IPython : 7.24.1
             scipy : 1.7.0
              tqdm : 4.61.1
--------------------------------------------------------------------------------
```

**Comment by kymok** [comment]
For now perhaps you could use following workaround:

```py
def add_redundant_vertex(mesh):
    # points
    points = np.array(mesh.points)
    last_point = points[len(arr)-1]

    # construct vtkPoints and duplicate last point
    vtk_pts = vtk.vtkPoints()
    for point in points:
        vtk_pts.InsertNextPoint(point)
    vtk_pts.InsertNextPoint(last_point)

    # set points and return result
    mesh_copy = mesh.copy()
    mesh_copy.SetPoints(vtk_pts)
    return mesh_copy
```

**Comment by akaszynski** [comment]
Thanks for pointing this out.  It's being addressed in #1454 and we'll have this patched in the next minor release (probably two weeks).

**Comment by kymok** [comment]
Thank you for quick response!
